src/runtime_src/core/tools/common/tests/TestNPU3Throughput.cpp

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "TestNPU3Throughput.h"
+#include "tools/common/XBUtilities.h"
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_hw_context.h"
+#include "xrt/xrt_kernel.h"
+#include "experimental/xrt_elf.h"
+#include "experimental/xrt_ext.h"
+#include "experimental/xrt_module.h"
+namespace XBU = XBUtilities;
+
+#include <filesystem>
+
+static constexpr size_t host_app = 1; //opcode
+static constexpr size_t buffer_size = 20;
+static constexpr int itr_count_throughput = 2500;
+// ----- C L A S S   M E T H O D S -------------------------------------------
+TestNPU3Throughput::TestNPU3Throughput()
+  : TestRunner("throughput", "Run end-to-end throughput test")
+{}
+
+boost::property_tree::ptree
+TestNPU3Throughput::run(std::shared_ptr<xrt_core::device> dev)
+{
+  boost::property_tree::ptree ptree = get_test_header();
+  ptree.erase("xclbin");
+  ptree.erase("elf");
+
+  const auto xclbin_name = xrt_core::device_query<xrt_core::query::xclbin_name>(dev, xrt_core::query::xclbin_name::type::nop);
+  auto xclbin_path = findPlatformFile(xclbin_name, ptree);
+  if (!std::filesystem::exists(xclbin_path))
+    return ptree;
+
+  logger(ptree, "Xclbin", xclbin_path);
+
+  xrt::xclbin xclbin;
+  try {
+    xclbin = xrt::xclbin(xclbin_path);
+  }
+  catch (const std::runtime_error& ex) {
+    logger(ptree, "Error", ex.what());
+    ptree.put("status", test_token_failed);
+    return ptree;
+  }
+
+  //Elf flow
+  const auto elf_name = xrt_core::device_query<xrt_core::query::sequence_name>(dev, xrt_core::query::sequence_name::type::nop);
+  auto elf_path = findPlatformFile(elf_name, ptree);
+  if (!std::filesystem::exists(elf_path))
+    return ptree;
+
+  logger(ptree, "Elf", elf_path);
+
+  xrt::elf elf;
+  try {
+    elf = xrt::elf(elf_path);
+  }
+  catch (const std::runtime_error& ex) {
+    logger(ptree, "Error", ex.what());
+    ptree.put("status", test_token_failed);
+    return ptree;
+  }
```

Comment:
From what I can tell, this is the only part which is different from previous devices. Is there a way we can use the older host code and trigger this part only when npu3 is detected? You can probably query the device name and see if it has "npu3" in it. 
```
if(xrt_core::device_query<xq::rom_vbnv>(device).find("npu3") != std::string::npos)
    //run elf stuff
```
I'm not fond of the device name search thing, but we'll have better way of managing this in the future. For now this should be ok

Reply:
@AShivangi the npu3 has different device type as UMQ.
https://github.com/amd/xdna-driver/blob/f158f2e5ab27ce21ed7f488823d4fe33dc8597b3/src/include/uapi/drm_local/amdxdna_accel.h#L60

However, I think the KMQ might be switched to new flow as well in the future.
We can temporary use the device type UMQ for NPU3, and eventually all xmi should switch to elf flow?